packages/core/src/Stores/gtm-store.js

```diff
@@ -96,14 +96,127 @@ export default class GTMStore extends BaseStore {
                 chart           : {
                     toolbar_position: this.root_store.ui.is_chart_layout_default ? 'bottom' : 'left',
                     chart_asset_info: this.root_store.ui.is_chart_asset_info_visible ? 'visible' : 'hidden',
-                    // chart_type      : this.root_store.modules.smart_chart.chart_type,
-                    // granularity     : this.root_store.modules.smart_chart.granularity,
+                    chart_type      : this.root_store.modules.smart_chart.chart_type,
+                    granularity     : this.root_store.modules.smart_chart.granularity,
                 },
             },
         };
         this.pushDataLayer(data);
     }
 
+    /**
+     * Pushes deposit & withdrawal data from transaction-stream to GTM
+     *
+     * @param {object} response
+     * @param {object} extra_data
+     */
+    @action.bound
+    pushTransactionData(response, extra_data = {}) {
+        if (!this.is_gtm_applicable || this.root_store.client.is_virtual) return;
+        if (!response.transaction || !response.transaction.action) return;
+        if (!['deposit', 'withdrawal'].includes(response.transaction.action)) return;
+
+        const moment_now  = toMoment();
+        const storage_key = 'GTM_transactions';
+
+        // Remove values from prev days so localStorage doesn't grow to infinity
+        let gtm_transactions = JSON.parse(localStorage.getItem(storage_key)) || {};
+        if (Object.prototype.hasOwnProperty.call(gtm_transactions, 'timestamp')) {
+            if (moment_now.isAfter(epochToMoment(gtm_transactions.timestamp), 'day')) {
+                localStorage.removeItem(storage_key);
+                gtm_transactions = { timestamp: moment_now.unix() };
+            }
+        }
+        const transactions_arr = gtm_transactions.transactions || [];
+        if (!transactions_arr.includes(response.transaction.transaction_id)) {
+            const data = {
+                event           : 'transaction',
+                bom_account_type: this.root_store.client.account_type,
+                bom_today       : moment_now.unix(),
+                transaction     : {
+                    id     : response.transaction.transaction_id,
+                    type   : response.transaction.action,
+                    time   : response.transaction.transaction_time,
+                    amount : response.transaction.amount,
+                    balance: response.transaction.balance,
```

Comment:
What is transaction balance?

Reply:
@ashkanx This is the account balance left after transaction has streamed.

Reply:
please remove this. I don't think we need this